src/devices_models/devices/thermal_generation.jl

```diff
@@ -517,6 +563,28 @@ function calculate_aux_variable_value!(
 
     return
 end
+
+function calculate_aux_variable_value!(
+    optimization_container::OptimizationContainer,
+    key::AuxVarKey{PowerOutput, T},
+    system::PSY.System,
+) where {T <: PSY.ThermalGen}
+    devices = PSY.get_components(T, system)
+    time_steps = model_time_steps(optimization_container)
+    on_var_results = get_variable(optimization_container, OnVariable(), T)
+    p_var_results = get_variable(optimization_container, PowerAboveMinimumVariable(), T)
+    aux_var_container = get_aux_variables(optimization_container)[key]
```

Comment:
use the appropriate getter function with the var type and T

Reply:
I'm not sure what you mean, there are only two `get_variable` methods one uses VariableKey and the other is the one I use.